app/controllers/tags_controller.rb

```diff
@@ -142,6 +142,13 @@ def list
 
   Discourse.filters.each do |filter|
     define_method("show_#{filter}") do
+      synonym_tag = Tag.where_name(params[:tag_id]).where.not(target_tag_id: nil).first
```

Comment:
Since we only need the name of the target_tag, I think it is worthwhile to make this query cheaper by only fetching the target's tag name from the database instead of all the columns.